core/src/io/anuke/mindustry/graphics/BlockRenderer.java

```diff
@@ -242,29 +242,39 @@ public void processBlocks(){
     }
 
     public void drawBlocks(Layer stopAt){
-
+        int startIdx = iterateidx;
         for(; iterateidx < requestidx; iterateidx++){
+            BlockRequest request = requests.get(iterateidx);
 
-            if(iterateidx < requests.size && requests.get(iterateidx).layer.ordinal() > stopAt.ordinal()){
+            if (request.layer.ordinal() > stopAt.ordinal()){
                 break;
             }
 
-            BlockRequest req = requests.get(iterateidx);
-            Block block = req.tile.block();
-
-            if(req.layer == Layer.block){
-                block.draw(req.tile);
-                if(req.tile.entity != null && req.tile.entity.damaged()){
-                    block.drawCracks(req.tile);
-                }
-                if(block.synthetic() && req.tile.getTeam() != player.getTeam()){
-                    block.drawTeam(req.tile);
+            if (request.layer == Layer.power) {
+                if (iterateidx - startIdx > 0 && request.tile.pos() == requests.get(iterateidx - 1).tile.pos()) {
+                    continue;
                 }
-            }else if(req.layer == block.layer){
-                block.drawLayer(req.tile);
-            }else if(req.layer == block.layer2){
-                block.drawLayer2(req.tile);
             }
+            drawBlock(request);
+        }
+    }
+
+    private void drawBlock(BlockRequest req) {
```

Comment:
I don't think there's a good reason to put this in a separate method. If it's *not* called multiple times or overly long, making new methods just creates more code and increases confusion.

Reply:
Well, yeah I was going to make drawBlocks(Layer stopAt) cleaner. Will change that.